spyder/plugins/explorer/widgets/explorer.py

```diff
@@ -558,6 +648,80 @@ def open(self, fnames=None):
+        return_codes = {}
+        if app_path:
+            fnames = self.get_selected_filenames()
+
+            # Add quotes as needed
+            if sys.platform != 'darwin':
+                new_fnames = []
+                for fname in fnames:
+                    if ' ' in fname:
+                        if '"' in fname:
+                            fname = '"{}"'.format(fname)
+                        else:
+                            fname = "'{}'".format(fname)
+                        new_fnames.append(fname)
+                    else:
+                        new_fnames.append(fname)
+                fnames = new_fnames
+
+            if sys.platform == 'darwin':
+                cmd = ['open', '-a', app_path] + fnames
+                return_code = subprocess.call(cmd)
+                return_codes[' '.join(cmd)] = return_code
+            elif os.name == 'nt':
+                cmd = [app_path] + fnames
+                return_code = subprocess.call(cmd)
+                return_codes[' '.join(cmd)] = return_code
+            else:
+                entry = parse_linux_desktop_entry(app_path)
+                app_path = entry['exec']
+                multi = []
+                extra = []
+                if len(fnames) == 1:
+                    fname = fnames[0]
+                    if '%u' in app_path:
+                        cmd = app_path.replace('%u', fname)
+                    elif '%f' in app_path:
+                        cmd = app_path.replace('%f', fname)
+                    elif '%U' in app_path:
+                        cmd = app_path.replace('%U', fname)
+                    elif '%F' in app_path:
+                        cmd = app_path.replace('%F', fname)
+                    else:
+                        cmd = app_path
+                        extra = fnames
+                elif len(fnames) > 1:
+                    if '%U' in app_path:
+                        cmd = app_path.replace('%U', ' '.join(fnames))
+                    elif '%F' in app_path:
+                        cmd = app_path.replace('%F', ' '.join(fnames))
+                    if '%u' in app_path:
+                        for fname in fnames:
+                            multi.append(app_path.replace('%u', fname))
+                    elif '%f' in app_path:
+                        for fname in fnames:
+                            multi.append(app_path.replace('%f', fname))
+                    else:
+                        cmd = app_path
+                        extra = fnames
+
+                if multi:
+                    for cmd in multi:
+                        return_code = subprocess.call([cmd], shell=True)
+                        return_codes[cmd]= return_code
+                else:
+                    return_code = subprocess.call([cmd] + extra, shell=True)
+                    return_codes[cmd]= return_code
+
+        # TODO: Handle code. Display message if problem?
+        print(return_codes)

```

Comment:
This `print` is probably causing our tests to fail. And please address the TODO you left on the line above.